File: exercises/beecrowd/1018/1018.go
// https://judge.beecrowd.com/pt/problems/view/1018
package main

import (
	"fmt"
)

func main() {
	var value int

	fmt.Scanln(&value)

	if value > 1000000 {
		return
	}

	result := map[int]int{
		100: 0,
		50:  0,
		20:  0,
		10:  0,
		5:   0,
		2:   0,
		1:   0,
	}

	moneyNoteOrderPriority := []int{100, 50, 20, 10, 5, 2, 1}

	i := value
	for i > 0 {
		for _, v := range moneyNoteOrderPriority {
			if (i - v) >= 0 {
				i = i - v
				result[v]++
				break
			}
		}
	}

	fmt.Println(value)
	for _, v := range moneyNoteOrderPriority {
		fmt.Printf("%v nota(s) de R$ %v,00\n", result[v], v)
	}
}

File: exercises/beecrowd/1018/1018.ts
// https://judge.beecrowd.com/pt/problems/view/1018
import * as fs from "fs";

function main(): void {
	const input = fs.readFileSync("/dev/stdin", "utf8");
	const lines = input.split(/\r?\n/);
  
	const value = Number(lines.shift());
  
	if (value > 1000000) {
		return;
	}
  
	const result: Record<number, number> = {
		100: 0,
		50: 0,
		20: 0,
		10: 0,
		5: 0,
		2: 0,
		1: 0,
	};
  
	const moneyNoteOrderPriority: number[] = [100, 50, 20, 10, 5, 2, 1];
  
	let i = value;
	while (i > 0) {
		for (const v of moneyNoteOrderPriority) {
			if (i - v >= 0) {
				i -= v;
				result[v]++;
				break;
			}
		}
	}
	
	console.log(value);
	for (const v of moneyNoteOrderPriority) {
		console.log(`${result[v]} nota(s) de R$ ${v},00`);
	}
}

main()
